src/leejaeyeong/수학/BOJ_10343_행복한소수.java

```diff
@@ -0,0 +1,67 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+public class BOJ_10343_행복한소수 {
+	public static void main(String[] args) throws Exception {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		StringBuilder sb = new StringBuilder();;
+		StringTokenizer st;
+		
+		int P = stoi(br.readLine());
+		for (int i = 0; i < P; i++) {
+			st = new StringTokenizer(br.readLine());
+			int tc = stoi(st.nextToken());
+			int num = stoi(st.nextToken());
+			sb.append(tc + " " + num + " ");
+			if (isPrime(num) && isHappyNum(num)) {
+				sb.append("YES\n");
+				continue;
+			}
+			sb.append("NO\n");
+		}
+		System.out.println(sb.toString());
+	}
+
+	private static boolean isHappyNum(int num) {
+		Set<Integer> numSet = new HashSet<>();
+		while(true) {
+			int nextNum = compute(num);
+			if (nextNum == 1) return true; // 행복한 수인 경우
+			if (numSet.contains(nextNum)) return false; // 이미 계산한 수인 경우
+			numSet.add(nextNum);
+			num = nextNum;
+		}
+	}
+	
+	/* 각 자리수의 제곱을 더하는 함수 */
+	private static int compute(int num) {
+		String strNum = Integer.toString(num);
+		int sum = 0;
+		for (int i = 0; i < strNum.length(); i++) {
+			int n = stoi(Character.toString(strNum.charAt(i))); // 각 자리의 수
+			sum += n * n;
+		}
+		return sum;
+	}
+
+	/* 소수 판별 함수 */
+	private static boolean isPrime(int num) {
+		if (num == 1) 
+			return false;
+		
+		int divisor = (int)Math.sqrt((double)num); // 약수가 될 수 있는 최대 값
+		
+		for (int i = 2; i <= divisor; i++) {
```

Comment:
```suggestion
              for (int i = 2; i * i <= num; i++) {
```
로 수정하여  `Math.sqrt()`를 쓰지 않고도 divisor를 구할 수 있습니다.

Reply:
이 방법이 더 깔끔하네요 반영하겠습니다!